### PR TITLE
adds gummykey

### PR DESCRIPTION
**Pull request by gumorr**
## Description
adds support for a keyboard i built called the "GummyKey". forgive me if theres any glaring issues, this is my first time doing something like this.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/gummykey/config.h [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ * Keyboard Matrix Assignments
+ *
+ * Change this to how you wired your keyboard
+ * COLS: AVR pins used for columns, left to right
+ * ROWS: AVR pins used for rows, top to bottom
+ * DIODE_DIRECTION: COL2ROW = COL = Anode (+), ROW = Cathode (-, marked on diode)
+ *                  ROW2COL = ROW = Anode (+), COL = Cathode (-, marked on diode)
+ *
+*/
+
```
Comment (waffle87) [inline keyboards/gummykey/config.h:23]:
```suggestion
```

File: keyboards/gummykey/info.json [hunk 2]
```diff
@@ -0,0 +1,86 @@
+{
+    "keyboard_name": "GummyKey",
+    "manufacturer": "Gumorr",
+    "url": "https://github.com/gumorr/GummyKey",
+    "maintainer": "Gumorr",
+    "usb": {
+        "vid": "0xAA12",
+        "pid": "0x0001",
+        "device_version": "1.0.0"
+    },
```
Comment (waffle87) [inline keyboards/gummykey/info.json:10]:
```suggestion
    },
    "processor": "atmega32u4",
    "bootloader": "caterina",
```

File: keyboards/gummykey/keymaps/default/readme.md [hunk 3]
```diff
@@ -0,0 +1 @@
+# Default layout for GummyKey
```
Comment (waffle87) [inline keyboards/gummykey/keymaps/default/readme.md:1]: This file should be removed.

File: keyboards/gummykey/readme.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# GummyKey
+
+![gummykey](https://user-images.githubusercontent.com/102767971/215166168-12f08a78-ce6e-4368-9a70-0d4e8173b8e2.png)
```
Comment (waffle87) [inline keyboards/gummykey/readme.md:3]: This image should be rehosted on Imgur.

File: keyboards/gummykey/readme.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+a split keyboard i made
+
+* Keyboard Maintainer: [Gummor](https://github.com/gumorr)
+* Hardware Supported: Pro Micro
+* Hardware Availability: [Github](https://github.com/gumorr/GummyKey)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make gummykey:default
+
```
Comment (waffle87) [inline keyboards/gummykey/readme.md:14]:
```suggestion

* Keyboard Maintainer: [Gummor](https://github.com/gumorr)
* Hardware Supported: Pro Micro
* Hardware Availability: [Github](https://github.com/gumorr/GummyKey)

Make example for this keyboard (after setting up your build environment):

    make gummykey:default

Flashing example for this keyboard:

    make gummykey:default:flash
    
```

File: keyboards/gummykey/rules.mk [hunk 6]
```diff
@@ -0,0 +1,19 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = caterina
+
```
Comment (waffle87) [inline keyboards/gummykey/rules.mk:6]:
```suggestion
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/gummykey/rules.mk [hunk 7]
```diff
@@ -0,0 +1,13 @@
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
```
Comment (zvecr) [inline keyboards/gummykey/rules.mk:8]:
```suggestion
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no        # Commands for debug and configuration
```

File: keyboards/gummykey/readme.md [hunk 8]
```diff
@@ -0,0 +1,20 @@
+# GummyKey
+
+![gummykey](https://i.imgur.com/R6ffs2B.png)
+
+a split keyboard i made
```
Comment (zvecr) [inline keyboards/gummykey/readme.md:5]:
```suggestion
A 4x6+5 split keyboard.
```

File: keyboards/gummykey/readme.md [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+Make example for this keyboard (after setting up your build environment):
+
+    make gummykey:default
+
+Flashing example for this keyboard:
+
+	make gummykey:default:flash
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information.
+Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```
Comment (zvecr) [inline keyboards/gummykey/readme.md:20]: Missing <https://github.com/qmk/qmk_firmware/blob/master/data/templates/keyboard/readme.md#bootloader> per template.

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/gummykey/readme.md [hunk 10]
```diff
@@ -0,0 +1,27 @@
+# GummyKey
+
+![gummykey](https://i.imgur.com/R6ffs2B.png)
```
Comment (drashna) [inline keyboards/gummykey/readme.md:3]:
```suggestion
![gummykey](https://i.imgur.com/R6ffs2Bh.png)
```

File: keyboards/gummykey/config.h [hunk 11]
```diff
@@ -0,0 +1,33 @@
+// Copyright 2023 Gummor (@gumorr)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include "config_common.h"
```
Comment (drashna) [inline keyboards/gummykey/config.h:6]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/gummykey/config.h [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+#define MATRIX_COL_PINS { F5, F6, F7, B1, B3, B2 }
+
+#define USE_I2C
+
+/* COL2ROW, ROW2COL, or CUSTOM_MATRIX */
+#define DIODE_DIRECTION ROW2COL
+
+/* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
+#define DEBOUNCE 5
+
```
Comment (waffle87) [inline keyboards/gummykey/config.h:17]:
```suggestion
```

File: keyboards/gummykey/info.json [hunk 13]
```diff
@@ -0,0 +1,88 @@
...
+    "manufacturer": "Gumorr",
+    "url": "https://github.com/gumorr/GummyKey",
+    "maintainer": "Gumorr",
+    "usb": {
+        "vid": "0xAA12",
+        "pid": "0x0001",
+        "device_version": "1.0.0"
+	},
+	"processor": "atmega32u4",
+	"bootloader": "caterina",
```
Comment (waffle87) [inline keyboards/gummykey/info.json:12]:
```suggestion
    },
    "processor": "atmega32u4",
    "bootloader": "caterina",
    "diode_direction": "ROW2COL",
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Review by waffle87 — APPROVED** [review]

**Comment by gumorr** [comment]
is this gonna be merged any time soon?

**Review by drashna — APPROVED** [review]

**Pull request merged**